packages/core/src/app/customer/Customer.tsx

```diff
@@ -393,195 +208,167 @@ class Customer extends Component<CustomerProps & WithCheckoutCustomerProps & Ana
+                onCancel={handleCancelSignIn}
+                onChangeEmail={handleChangeEmail}
+                onContinueAsGuest={executePaymentMethodCheckoutOrContinue}
+                onCreateAccount={showCreateAccount}
+                onSendLoginEmail={handleEmailLoginClicked}
+                onSignIn={handleSignIn}
+                signInError={customerData.data.signInError}
+                viewType={viewType}
+            />
+        );
+    }, [state.draftEmail, customerData.data.email, customerData.data.isFloatingLabelEnabled, customerData.data.signInError, isEmbedded, viewType, handleCancelSignIn, handleChangeEmail, executePaymentMethodCheckoutOrContinue, showCreateAccount, handleEmailLoginClicked, handleSignIn]);
 
-        analyticsTracker.customerPaymentMethodExecuted(payload);
-    }
-}
+    // Main render logic
+    const shouldRenderGuestForm = viewType === CustomerViewType.Guest;
+    const shouldRenderCreateAccountForm = viewType === CustomerViewType.CreateAccount;
+    const shouldRenderLoginForm = !shouldRenderGuestForm && !shouldRenderCreateAccountForm;
 
-export function mapToWithCheckoutCustomerProps({
-    checkoutService,
-    checkoutState,
-}: CheckoutContextProps): WithCheckoutCustomerProps | null {
-    const {
-        data: {
-            getBillingAddress,
-            getCustomerAccountFields,
-            getCheckout,
-            getCustomer,
-            getCart,
-            getSignInEmail,
-            getConfig,
-            isPaymentDataRequired,
-        },
-        errors: { getSignInError, getSignInEmailError, getCreateCustomerAccountError },
-        statuses: {
-            isContinuingAsGuest,
-            isExecutingPaymentMethodCheckout,
-            isInitializingCustomer,
-            isSigningIn,
-            isSendingSignInEmail,
-            isCreatingCustomerAccount,
-        },
-    } = checkoutState;
-
-    const billingAddress = getBillingAddress();
-    const checkout = getCheckout();
-    const customer = getCustomer();
-    const cart = getCart();
-    const signInEmail = getSignInEmail();
-    const config = getConfig();
-
-    if (!checkout || !config || !cart) {
+    if (!state.isReady) {
         return null;
     }
 
-    const {
-        checkoutSettings: {
-            privacyPolicyUrl,
-            requiresMarketingConsent,
-            isSignInEmailEnabled,
-            isAccountCreationEnabled,
-            isExpressPrivacyPolicy,
-            shouldRedirectToStorefrontForAuth
-        },
-    } = config;
-
-    const providerWithCustomCheckout = getProviderWithCustomCheckout(
-        config.checkoutSettings.providerWithCustomCheckout,
+    return (
+        <>
+            {state.isEmailLoginFormOpen && renderEmailLoginLinkForm()}
+            {shouldRenderLoginForm && renderLoginForm()}
+            {shouldRenderGuestForm && renderGuestForm()}
+            {shouldRenderCreateAccountForm && renderCreateAccountForm()}
```

Comment:
Thought we should avoid function calls in the returned component.

Reply:
Yep you are correct, I am going to do all this as a follow up to this PR.